.github/workflows/gotest.yml

```diff
@@ -59,8 +65,38 @@ jobs:
           mv go.mod.bak go.mod
           mv go.sum.bak go.sum
         working-directory: docs/examples/kubo-as-a-library
-      - uses: actions/upload-artifact@v3
+      - name: Create a proper JUnit XML report
+        uses: pl-strflt/gotest-json-to-junit-xml@v1
+        with:
+          input: test/unit/gotest.json
+          output: test/unit/gotest.junit.xml
+        if: failure() || success()
```

Comment:
The JUnit XML report needs to be recreated because, as it turns out, the XML that `gotestsum` produces is not well-formatted (e.g. it doesn't use `package` attribute in `testsuite` which is needed for other tools to work properly).